src/Microsoft.AspNet.Routing/RouteBase.cs

```diff
@@ -237,11 +238,12 @@ public virtual VirtualPathData GetVirtualPath(VirtualPathContext context)
             }
         }
 
-        private void EnsureBinder()
+        private void EnsureBinder(HttpContext context)
         {
             if (_binder == null)
             {
-                _binder = new TemplateBinder(ParsedTemplate, Defaults);
+                var urlEncoder = context.RequestServices.GetRequiredService<UrlEncoder>();
+                _binder = new TemplateBinder(ParsedTemplate, urlEncoder, Defaults);
```

Comment:
Are you complaining about the `TemplateBinder` type or the need to `GetRequiredService`?